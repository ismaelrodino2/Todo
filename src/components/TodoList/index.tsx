import { useEffect, useState } from 'react';
import { useFilterValue } from '../../Store/filter-value';
import { TodoListItem } from '../TodoListItem';

interface TodoListProps {
  todos: any;
  toggleTodo: any;
  setTodos: any;
}

export const TodoList: React.FC<TodoListProps> = ({
  todos,
  toggleTodo,
  setTodos,
}) => {
  const value = useFilterValue((state) => state.value);
  console.log(value);

  const [checkboxesToDisplay, SetcheckboxesToDisplay] = useState(todos);

  useEffect(() => {
    if (value === '0') {
      SetcheckboxesToDisplay(
        todos.filter((todo: { complete: boolean; }) => todo.complete === true),
      );
    }
    if (value === '1') {
      SetcheckboxesToDisplay(
        todos.filter((todo: { complete: boolean }) => !todo.complete),
      );
    }
    if (value === '2') {
      SetcheckboxesToDisplay(todos);
    }
    console.log(checkboxesToDisplay);
    console.log(todos);
    console.log(typeof value);
  }, [todos, value]);

  return (
    <ul>
      {checkboxesToDisplay.map((todo:any) => {
        return (
          <TodoListItem key={todo.text} todo={todo} toggleTodo={toggleTodo} setTodos={setTodos} todos={todos} />
        );
      })}
    </ul>
  );

};
