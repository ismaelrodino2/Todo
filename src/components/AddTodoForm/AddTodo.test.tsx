import { AddTodoForm } from '.';
import { TodoList } from '../TodoList';

import { render, waitFor, fireEvent } from '@testing-library/react';

describe('Tests for Todo component', () => {
  it('Should add todo when form has been submited', async () => {
    const { getByTestId } = render(<AddTodoForm />);
    const { getByText: getList } = render(<TodoList />);
    const fieldNode = await waitFor(() => getByTestId('form-field'));
    const newTask = 'testing';
    fireEvent.change(fieldNode, { target: { value: newTask } });
    expect((fieldNode as HTMLInputElement).value).toEqual(newTask);

    const btnNode = await waitFor(() => getByTestId('form-btn'));
    fireEvent.click(btnNode);

    const listNode = await waitFor(() => getList(newTask));

    expect(listNode).toBeDefined()
    expect(getByTestId).toMatchSnapshot();

  });
});
