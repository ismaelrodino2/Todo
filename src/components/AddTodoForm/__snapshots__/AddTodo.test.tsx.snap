// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Todo component Should add todo when form has been submited 1`] = `[Function]`;
